app/hooks/useUserLocale.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+import WebApp from '@twa-dev/sdk'
+
+export const useUserLocale = () => {
+  const [userLocale, setUserLocale] = useState<string>()
+
+  useEffect(() => {
+    try {
+      // @telegram-apps/sdk-react doesn't provide access to the language code property
```

Comment:
You can access the `languageCode` directly from the user object, no need to install another library for that.
Reference: [telegram-apps docs](https://docs.telegram-mini-apps.com/packages/telegram-apps-sdk/1-x/init-data/user#languagecode)
You can test this using the `user` provided in the Telegram provider.

When using `user.languageCode`, make sure to check if it includes the full locale, on my side, it only returns `'en'`, while my browser returns `'en-GB'`. So ideally, give preference to the browser's language setting and fall back to Telegram’s `languageCode` if needed.